File: 1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const map = new Map<number, number>()

    for (let i = 0; i < nums.length; i++) {
        const num = nums[i]
        const neededNum = target - num

        if (map.has(neededNum)) return [map.get(neededNum), i]

        map.set(num, i)
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index_map = {}

        for i, num in enumerate(nums):
            if target - num in index_map:
                return [i, index_map[target - num]]
            index_map[num] = i

        